server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/change/MailboxChangeListenerTest.scala

```diff
@@ -66,171 +67,279 @@ class MailboxChangeListenerTest {
-
-    assertThat(repository.getSinceState(ACCOUNT_ID, state, None.toJava).block().getUpdated)
-      .isEmpty()
-  }
-
-  @Test
-  def deleteMessageFromMailboxShouldStoreUpdateEvent(): Unit = {
-    val mailboxSession = MailboxSessionUtil.create(BOB)
-    val path = MailboxPath.forUser(BOB, "test")
-    val inboxId: MailboxId = mailboxManager.createMailbox(path, mailboxSession).get
-    val messageManager: MessageManager = mailboxManager.getMailbox(inboxId, mailboxSession)
-    val appendResult: AppendResult = messageManager.appendMessage(AppendCommand.builder().build("header: value\r\n\r\nbody"), mailboxSession)
-
-    val state = stateFactory.generate()
-    repository.save(MailboxChange.builder().accountId(ACCOUNT_ID).state(state).date(ZonedDateTime.now).isCountChange(false).created(List[MailboxId](TestId.of(0)).asJava).build).block()
-    messageManager.delete(List(appendResult.getId.getUid).asJava, mailboxSession)
-
-    assertThat(repository.getSinceState(ACCOUNT_ID, state, None.toJava).block().getUpdated)
-      .containsExactly(inboxId)
-  }
-
-  @Test
-  def deleteMailboxNameShouldStoreDestroyedEvent(): Unit = {
-    val mailboxSession = MailboxSessionUtil.create(BOB)
-    val path = MailboxPath.forUser(BOB, "test")
-    val inboxId: MailboxId = mailboxManager.createMailbox(path, mailboxSession).get
-
-    val state = stateFactory.generate()
-    repository.save(MailboxChange.builder().accountId(ACCOUNT_ID).state(state).date(ZonedDateTime.now).isCountChange(false).created(List[MailboxId](TestId.of(0)).asJava).build).block()
-
-    mailboxManager.deleteMailbox(inboxId, mailboxSession)
-
-    assertThat(repository.getSinceState(ACCOUNT_ID, state, None.toJava).block().getDestroyed)
-      .containsExactly(inboxId)
+  @Nested
+  class EmailChangeEvents {
+    @Test
+    def appendMessageToMailboxShouldStoreCreatedEvent(): Unit = {
+      val mailboxSession = MailboxSessionUtil.create(BOB)
+      val path = MailboxPath.forUser(BOB, "test")
+      val inboxId: MailboxId = mailboxManager.createMailbox(path, mailboxSession).get
+
+      val state = stateFactory.generate()
+      emailChangeRepository.save(EmailChange.builder()
+          .accountId(ACCOUNT_ID)
+          .state(state)
+          .date(ZonedDateTime.now)
+          .isDelegated(false)
+          .created(TestMessageId.of(0))
+          .build)
+        .block()
+
+      mailboxManager.applyRightsCommand(path, MailboxACL.command().forUser(ALICE).rights(MailboxACL.Right.Read).asAddition(), mailboxSession)
+
+      val appendResult: AppendResult = mailboxManager
+        .getMailbox(inboxId, mailboxSession)
+        .appendMessage(AppendCommand.builder().build("header: value\r\n\r\nbody"), mailboxSession)
+
+      assertThat(emailChangeRepository.getSinceState(ACCOUNT_ID, state, None.toJava).block().getCreated)
+        .containsExactly(appendResult.getId.getMessageId)
+    }
+
+    @Test
+    def addFlagsShouldStoreUpdateEvent(): Unit = {
```

Comment:
In MailboxChange we only care about changes happen to the "SEEN" flag. This is not the case with EmailChange?

Reply:
No we care about all flags, because it changes the mail. We care of seen flag in mailbox because it changes the count, other flags don't change the mailbox. I confirmed it with benoit last week